CONTRIBUTING.md

```diff
@@ -0,0 +1,263 @@
+## Style guidelines
+
+We follow the guidelines proposed in the small book, [_Embedded C Coding
+Standard_](https://barrgroup.com/sites/default/files/barr_c_coding_standard_2018.pdf) by Michael
+Barr, with some exceptions. This standard was chosen because it is available for free and because
+it was similar in some superficial ways to the style that we already prefer.
+
+### Project-specific exceptions to the standard
+We use some arbitrary exceptions to the Barr standard. These serve
+* to minimize the version history noise required to come into compliance, and
+* to avoid unnecessary differences between our C style guidelines and our Java style guidelines.
+
+#### Braces (1.3)
+
+We use K&R-style braces. Example:
+```
+if (NULL == p_event) {
+    lf_print_error_and_exit("Out of memory!");
+}
+```
+
+#### The pointer operator * in declarations (3.1g)
+
+We only use a space on the variable name side (RHS) of *. Example:
+```
+int* ptr;
+```
+
+#### Function declarations (3.1j)
+
+We do not use the special case mentioned in Barr 3.1j.
```

Comment:
Instead of making the reader hunt this down, I suggest explaining:

I.e., we do not a space between the function name and the left argument parenthesis when defining the function.

Reply:
Done